File: Scripting-Engine/Scripting-Engine/LeagueSandbox-Default/Champions/Ezreal/W.lua
Vector2 = require 'Vector2' -- include 2d vector lib

function onStartCasting()
    addParticleTarget(owner, "ezreal_bow_yellow.troy", owner, 1, "L_HAND")
end

function onFinishCasting()
    local current = Vector2:new(owner.X, owner.Y)
    local to = (Vector2:new(spell.X, spell.Y) - current):normalize()
    local range = to * 1000
    local trueCoords = current + range
	
    addProjectile("EzrealEssenceFluxMissile", trueCoords.x, trueCoords.y)
end

function applyEffects()
    if owner.Team ~= getTarget().Team then
        dealMagicalDamage(25+spellLevel*45+owner:GetStats().AbilityPower.Total*0.8)
    end
end

function onUpdate(diff)
end

File: Scripting-Engine/Scripting-Engine/LeagueSandbox-Default/Champions/Ezreal/W.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Numerics;
using LeagueSandbox.GameServer.Logic.GameObjects;
using LeagueSandbox.GameServer.Logic.API;

namespace Ezreal
{
    class W
    {
        static void onStartCasting(Champion owner)
        {
            ApiFunctionManager.AddParticleTarget(owner, "ezreal_bow_yellow.troy", owner, 1, "L_HAND");
        }
        static void onFinishCasting(Champion owner, Spell spell)
        {
            Vector2 current = new Vector2(owner.X, owner.Y);
            Vector2 to = Vector2.Normalize(new Vector2(spell.X, spell.Y) - current);
            Vector2 range = to * 1000;
            Vector2 trueCoords = current + range;

            spell.AddProjectile("EzrealEssenceFluxMissile", trueCoords.X, trueCoords.Y);
        }
        static void applyEffects(Champion owner, Unit target, Spell spell)
        {
            if(owner.Team != target.Team)
            {
                owner.dealDamageTo(target, 25f + spell.Level * 45f + owner.GetStats().AbilityPower.Total * 0.8f, DamageType.DAMAGE_TYPE_MAGICAL, DamageSource.DAMAGE_SOURCE_SPELL, false);
            }
        }
        static void onUpdate(double diff) { }
    }
}
